src/AllocationIDTracker.sol

```diff
@@ -78,7 +84,9 @@ contract AllocationIDTracker {
             abi.encodePacked(sender, allocationID, msg.sender)
         );
         bytes32 digest = ECDSA.toEthSignedMessageHash(messageHash);
-        require(ECDSA.recover(digest, proof) == allocationID, "!proof");
+        if (ECDSA.recover(digest, proof) != allocationID) {
+            revert InvalidProof();
+        }
         return true;
```

Comment:
The return could be removed from this function since we are not using it anywhere and it's a private func.